src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java

```diff
@@ -281,8 +281,13 @@ public void onSubscribe(Flow.Subscription subscription) {
 
         @Override
         public void onNext(List<ByteBuffer> items) {
+            int size = Utils.remaining(items, Integer.MAX_VALUE);
             try {
-                out.write(items.toArray(Utils.EMPTY_BB_ARRAY));
+                long bytesWritten = 0;
+                ByteBuffer[] buffers = items.toArray(Utils.EMPTY_BB_ARRAY);
+                while (bytesWritten < size) {
+                    bytesWritten += out.write(buffers);
```

Comment:
Should we do something special here if no bytes are written, to avoid a busy loop?

Reply:
> Should we do something special here if no bytes are written, to avoid a busy loop?

I wonder how / if this could occur in practice? From what I see elsewhere the simple, "don't check no bytes written", loop is common.  ( It shouldn't be so hard/dangerous to use these gathering write APIs :-( )

Reply:
I would expect that 0 can only be returned if the channel is configured non blocking. So maybe for our piece of mind we could throw an IOException if that happens, which would then be caught below and cause close() and subscription.cancel() to be called, and the result to be completed exceptionally?